rmgpy/reaction.py

```diff
@@ -1123,3 +990,62 @@ def copy(self):
         
         return other
 
+def _isomorphicSpeciesList(list1, list2, checkIdentical=False):
+    """
+    This method compares whether lists of species or molecules are isomorphic
+    or identical. It is used for the 'isIsomorphic' method of Reaction class.
+    It likely can be useful elswehere as well:
+        
+        list1 - list of species/molecule objects of reaction1
+        list2 - list of species/molecule objects of reaction2
+        checkIdentical - if true, uses the 'isIdentical' comparison
+                         if false, uses the 'isIsomorphic' comparison
+                         
+    Returns True if the lists are isomorphic/identical & false otherwise
+    """
+
+    def comparison_method(other1, other2, checkIdentical=checkIdentical):
+        if checkIdentical:
+            return other1.isIdentical(other2)
+        else:
+            return other1.isIsomorphic(other2)
+
+    if len(list1) == len(list2) == 1:
+        if comparison_method(list1[0], list2[0]):
+            return True
+    elif len(list1) == len(list2) == 2:
+        if comparison_method(list1[0], list2[0]) \
+                    and comparison_method(list1[1], list2[1]):
+            return True
+        elif comparison_method(list1[0], list2[1]) \
+                    and comparison_method(list1[1], list2[0]):
+            return True
+    elif len(list1) == len(list2) == 3:
```

Comment:
Wouldn't it be much faster and shorter to iterate through the lists and pop out entries that are the same from both lists as you go.  I believe the below code would reduce the maximum calls to comparison_method to 6 rather than the current 18.  
list1Temp = []
list2Temp = []
indexList = []
k = 0
while list1 != []:
   entry = list1[k]
   for i,entry2 in enumerate(list2):
       if comparison_method(entry,entry2):
           temp = list1.pop(k)
           list1temp.append(temp)
           temp = list2.pop(i)
           list2temp.append(temp)
           indexList.append(i)
           break
   else: #loop doesn't break, indexList is missing entries
         for j in xrange(len(indexList),len(list2)):
             indexList.append(j)
         list2 = [list2[i] for i in indexList]
         return False
   #everything is ok, list1 is fine, rebuild list2
   list2 = [list2[i] for i in indexList]
   return True